### ISSUE TITLE
NgIf doesn't work properly in case of async event 

### ISSUE THREAD

**Issue by marcoCatto** [angular/angular#46320]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

I have an ngIf controlling the visibility of a DOM element. For a moment the element is visible even if the ngIf should hide it. 

In the test, the ngIf shows either a div with "hello word" text or the _MyComponentComponent_ component. A promise in the OnInit of the main component shows _MyComponentComponent_ when resolved.

In the onInit of  _MyComponentComponent_, I check if the "hello world" div is still visible. The check returns true, while it should return false. With a breakpoint in the onInit of _MyComponentComponent_ it is possible to see that both the "hello world" div and _MyComponentComponent_ are rendered, while just one of the two should be.

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/angular-ivy-kzusn8?embed=1&file=src/app/my-component/my-component.component.ts

### Anything else?

The bug is present when an async event is thrown, not only with a promise.

**Comment by JoostK** [comment]
I am confused by the StackBlitz; there is no element with the `component` class name yet that is what is being queried, and `document.querySelectorAll` always returns a `NodeList` so testing for `!== undefined` is always true. If I change the condition to `.length > 0` then I get false, even if I change the queried class to `mainComponent` (which does exist).

tldr; this doesn't reproduce for me with the given StackBlitz.

**Comment by jessicajaniuk** [comment]
I've been playing in the stackblitz, too, and I've not been able to replicate the issue either. I've updated it to setTimeout and I see the delayed switching of templates as intended. It seems to be behaving like it should.
